### PR TITLE
created web application firewall module and enabled https on azure functions

### PR DESCRIPTION
**Pull request by adityapawan-b**
please do review the code. i have successfully deployed the frontdoor waf on ops env

### CODE REVIEW

File: services/fn/premium/base/v1.3/main.tf [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+# This will deploy an Azure Function to the target Resource Group / App Service Plan
+resource "azurerm_function_app" "function_app" {
+  name                      = var.service_settings.name
+  location                  = var.context.location
+  resource_group_name       = var.context.resource_group_name
+  app_service_plan_id       = var.service_settings.plan_id
+  storage_connection_string = var.service_settings.storage_account.connection_string
+  version                   = var.service_settings.runtime_version
+  https_only                = true
```
Comment (markti-persistent) [inline services/fn/premium/base/v1.3/main.tf:20]: if this is the only change, don't need to make a separate version v1.3, just update to v1.2

File: services/frontdoor/policy/base/v1/main.tf [hunk 2]
```diff
@@ -0,0 +1,31 @@
+resource "azurerm_frontdoor_firewall_policy" "waf" {
+  name                              = "${var.service_settings.name}"
+  resource_group_name               = var.context.resource_group_name
+  enabled                           = true
+
+  managed_rule {
```
Comment (markti-persistent) [inline services/frontdoor/policy/base/v1/main.tf:6]: can't there be more than one managed rule? what happens if we want to support more managed ruleset?
maybe create a version of the module where we deploy all the managed rulesets or alternatively, allow to pass in a array of managed rulesets you want to enable.

File: services/frontdoor/policy/base/v1/variables.tf [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+variable "managed_rules_settings" {
+    type = object({
+        type    = string 
+        version = string 
+    })
+}
+
+#variable "custom_rules_settings" {
```
Comment (markti-persistent) [inline services/frontdoor/policy/base/v1/variables.tf:37]: remove commented code

File: services/frontdoor/policy/base/v1/main.tf [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  name                              = "${var.service_settings.name}"
+  resource_group_name               = var.context.resource_group_name
+  enabled                           = true
+
+  managed_rule { 
+        type    = var.managed_rules_settings.type
+        version = var.managed_rules_settings.version
+      }
+  
+  #custom_rule {
```
Comment (markti-persistent) [inline services/frontdoor/policy/base/v1/main.tf:11]: remove commented code

File: services/frontdoor/policy/base/v1/variables.tf [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+}
+variable "service_settings" {
+  type = object({
+
+    name  = string
+
+  })
+}
+
+variable "policy_settings" {
```
Comment (markti-persistent) [inline services/frontdoor/policy/base/v1/variables.tf:20]: remove unused variables

File: services/frontdoor/policy/base/v1/variables.tf [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    type = object({
+        custom_block_response_status_code   = number 
+        custom_block_response_body          = string 
+        mode                                = string
+        redirect_url                        = string
+
+    })
+}
+
+variable "managed_rules_settings" {
```
Comment (markti-persistent) [inline services/frontdoor/policy/base/v1/variables.tf:30]: modify to support multiple managed rule sets or hard code a default configuration

File: services/fn/premium/base/v1.3/main.tf (hunk 1 above)
Comment (adityapawan-b) [reply services/fn/premium/base/v1.3/main.tf:20]: sure will change it

File: services/frontdoor/policy/base/v1/main.tf (hunk 2 above)
Comment (adityapawan-b) [reply services/frontdoor/policy/base/v1/main.tf:6]: i will look into the suggestion re-work accordingly

File: services/frontdoor/policy/base/v1/variables.tf (hunk 3 above)
Comment (adityapawan-b) [reply services/frontdoor/policy/base/v1/variables.tf:37]: this is for custom rules, if we like to define in future. as there are some user stories coming up on IP restrictions and geo filtering. However i will remove it

File: services/frontdoor/policy/base/v1/main.tf (hunk 4 above)
Comment (adityapawan-b) [reply services/frontdoor/policy/base/v1/main.tf:11]: this is for custom rules, if we like to define in future. as there are some user stories coming up on IP restrictions and geo filtering. However i will remove it

File: services/frontdoor/policy/base/v1/variables.tf (hunk 5 above)
Comment (adityapawan-b) [reply services/frontdoor/policy/base/v1/variables.tf:20]: sure will remove the unused variables

**Pull request closed without merging**